src/stateManager/StateManager.ts

```diff
@@ -357,15 +381,20 @@ class StateManager {
                 reduceData.encodedStateMachineState,
                 reduceData.joinChannelBlocks
             );
+
         const reducedForkId = ethers.keccak256(
             Codec.encode(snapshotData, Type.SnapshotData)
         );
 
+        // Update local state to the reduced fork
+        this.logger.debug(
+            `Reduction complete: transitioning to fork ${reducedForkId}`
+        );
         this.setGenesisState(
             snapshotData,
             encodedStateMachineState,
             reducedForkId,
-            genesisTimestamp,
+            Clock.getTimeInSeconds(),
```

Comment:
We can't do this - this has to be an objective timestamp, since it goes into the snapshot and the snapshot needs to be the same for everyone. That's why the timestamp is passed around since it originates from the chain